corehq/apps/users/forms.py

```diff
@@ -934,14 +939,17 @@ def __init__(self, project, request_user, *args, **kwargs):
                             disabled_phone=_("Setting a password is disabled. "
                                             "The user will set their own password on confirming "
                                             "their account phone number."),
+                            short=_(f"Your password should have at least {settings.MINIMUM_PASSWORD_LENGTH} \
+                            characters."),
                         )),
                         required=True,
                     ),
                     data_bind='''
                         css: {
                             'has-success': $root.passwordStatus() === $root.STATUS.SUCCESS,
                             'has-warning': $root.passwordStatus() === $root.STATUS.WARNING,
-                            'has-error': $root.passwordStatus() === $root.STATUS.ERROR,
+                            'has-error': $root.passwordStatus() === $root.STATUS.ERROR ||
+                             $root.passwordSatisfyLength() === false,
```

Comment:
I think we likely need to revisit how this is working. The intent here seems to be that the css will only ever bind to one of these options, but by adding in the length requirements, that no longer has to be true. For example, if a sufficiently complex password was created that wasn't long enough, this element would be tagged with `has-success` as well as `has-error`.

Reply:
Hi Matt, I add `$root.passwordSatisfyLength() === true` to `has-success` and `has-warning`, which will make all three condition exclusive.